backup/manager.go

```diff
@@ -241,13 +249,66 @@ func (b *Manager) IsSafeToRunNode(nodeID string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
+
 	for _, s := range snapshots {
 		if s.NodeID == nodeID && s.BackupID != "" && backupID != s.BackupID {
 			b.log.Errorf("remote restore was found for node %v.", nodeID)
 			b.log.Errorf("current backupID=%v, remote backupID-%v", backupID, s.BackupID)
 			return false, nil
 		}
 	}
+
+	return true, nil
+}
+
+// ensureLatestBackup checks whenever a given local state is newer than our snapshot.
+// if there is no local state, it returns true.
+func (b *Manager) EnsureLatestBackup(nodeID string) (bool, error) {
+
+	provider := b.GetProvider()
+	if provider == nil {
+		return true, ErrorNoProvider
+	}
+	snapshots, err := provider.AvailableSnapshots()
+	if err != nil {
+		return true, err
+	}
+
+	providerName, err := b.getProviderName()
+	if err != nil {
+		return true, err
+	}
+	b.log.Infof("Using provider: %v", providerName)
+
+	var snapshot SnapshotInfo
+	for _, s := range snapshots {
+		if s.NodeID == nodeID {
+			snapshot = s
+		}
+	}
+
+	if snapshot.ModifiedTime.IsZero() {
+		return true, nil
+	}
+
+	latestBackup, err := b.db.latestBackupTime(providerName)
+	if err != nil {
+		b.log.Errorf("error getting latestBackup exited with err: %v", err)
+		if !strings.Contains(err.Error(), "no backup time found") {
+			return true, err
+		}
+	}
+	if latestBackup.IsZero() {
+		b.log.Infof("Latest modified time is not set yet, setting it now.")
+		b.db.setLatestBackupTime(snapshot.ModifiedTime, b.providerName)
+		return true, nil
+	}
+
+	if snapshot.ModifiedTime.Before(latestBackup) {
```

Comment:
This checks if the cloud backup is BEFORE the local backup (last backup) right?
Which means the local backup is newer and not older than the cloud backup.

Reply:
Yes, we check if the local one is older, it can from time to time be newer than the cloud one due.